pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -1070,6 +1072,92 @@ struct ShapeOpConversion : public OpConversionPattern<ShapeOp> {
   }
 };
 
+struct ScatterOpConversion : public OpConversionPattern<ScatterOp> {
+  using OpConversionPattern<ScatterOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ScatterOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    IVLOG(2, "ScatterOpConversion::matchAndRewrite>");
+    // Helpful explanation of scatter from tensorflow docs:
+    // https://www.tensorflow.org/api_docs/python/tf/scatter_nd
+
+    auto loc = op.getLoc();
+    auto ctx = rewriter.getContext();
+    TypeConverter typeConverter;
+
+    // Create an adaptor, to interpret the operands
+    ScatterOpAdaptor adaptor(operands);
+    // 'tensor' provides update values
+    // 'dims' contains the destination indices
+    // 'other' is the shape of the output
+    // this is redundant because the result type also specifies output shape
+    auto updates = adaptor.tensor();
+    auto indices = adaptor.dims();
+
+    // Make an allocation for the output
+    auto resultType = typeConverter.convertType(op.result().getType());
+    auto resultMemRefType = resultType.cast<MemRefType>();
+    auto resultMemRef =
+        rewriter.create<AllocOp>(loc, resultMemRefType).getResult();
+
+    // Get the shape of the update tensor and create a parallel loop over its
+    // indexes; we will load each value from the updates, load its destination
+    // from the indexes, and store the value to the result.
+    auto updatesType = typeConverter.convertType(updates.getType());
```

Comment:
I would expect the type of the adaptor operands to already have been converted, so why do we need to convert it manually here?